scio-test/src/main/scala/com/spotify/scio/testing/PipelineTestUtils.scala

```diff
@@ -131,3 +140,42 @@ trait PipelineTestUtils {
   }
 
 }
+
+trait PipelineUnitTestUtils extends PipelineTestUtils with UnitTestContextProvider
+
+/** Internal trait for integration testing. */
+private[scio] trait PipelineITUtils extends PipelineTestUtils with IntegrationTestContextProvider {
+
+  private val gcsUtil = new GcsUtilFactory().create(this.context.options)
+
+  /**
+   * Test pipeline components with a [[ScioContext]] by supplying data for a distributed cache.
+   * @param data the data for the distributed cache
+   * @param path the GCS file where the cache will be written
+   * @param fn code that tests the components and verifies the result
+   *
+   * {{{
+   * runWithDistCache(Seq("a", "b"), "gs://<bucket>/<file>") { sc =>
+   *   val cache = sc.distCache("gs://<bucket>/<file>") (initFn) // Can use in test
+   *   sc.parallelize(Seq(1, 2, 3)).sum should containSingleValue (6)
+   * }
+   * }}}
+   */
+  def runWithDistCache[T1: ClassTag, T2: ClassTag](data: Iterable[T1], path: String)
```

Comment:
`T1` and `T2` aren't logical siblings so I'd use different letters, like `T` and `U`.